### PR TITLE
Add test for offline mode

### LINKED ISSUE #2622: test offline gmail mode - decrypt
here's a 1st priority issue for you! :)

Please test decrypting message in offline mode. It's supposed to be able to decrypt messages (at least the inline ones) without going online.

It's very important for a potential customer.

### PR DESCRIPTION
**Pull request by limonte**
Closes #2622 

@tomholub first of all, please give your feedback on the test. I decided to add the Gmail test to simulate the real use-case, but if you think that's not the best way, let me know.

Secondly, if the first part is alright and Gmail test is fine here, I'm struggling to get the `pgp_block.htm` frame, for some reason the `getFrame()` method can't find it and throwing the `Error: Frame not found: pgp_block.htm` exception.

### CODE REVIEW

File: test/source/tests/tests/gmail.ts [hunk 1]
```diff
@@ -116,6 +116,19 @@ export const defineGmailTests = (testVariant: TestVariant, testWithBrowser: Test
...
+      const gmailPage = await BrowserRecipe.openGmailPage(t, browser);
+      await gmailPage.type('[aria-label="Search mail"]', 'encrypted + signed with gpg');
+      await gmailPage.press('Enter'); // submit search
+      await gmailPage.page.waitFor(1000); // wait for search results
+      await gmailPage.page.setOfflineMode(true); // go OFFLINE
+      await gmailPage.press('Enter'); // open the message
+      // getFramesUrls will return one iframe which is expected
+      const urls = await gmailPage.getFramesUrls(['/chrome/elements/pgp_block.htm'], { sleep: 1 });
+      // but getFrame will fail, why?
+      const pgpBlockPage = await gmailPage.getFrame(['pgp_block.htm']);
```
Comment (tomholub) [inline test/source/tests/tests/gmail.ts]: Ah, puppeteer bug https://github.com/puppeteer/puppeteer/issues/2548

The solution is to copy the url and open in separate frame. At which point.. you might as well make that a mock test where you hard code the url.. like other decrypt tests - unfortunately.

Comment (limonte) [reply test/source/tests/tests/gmail.ts]: > Ah, puppeteer bug puppeteer/puppeteer#2548

Thanks, subscribed :+1:

Comment (limonte) [reply test/source/tests/tests/gmail.ts]: > At which point.. you might as well make that a mock test where you hard code the url.. like other decrypt tests - unfortunately.

Yeah, that kinda ruins the point of the test. What do you think about using a visual test in this case? I'm talking about taking the screenshot of `iframe.pgp_block` and comparing it to the expected one. This is the result of `ElementHandle.screenshot`:

Comment (tomholub) [reply test/source/tests/tests/gmail.ts]: That I think would be too too noisy - when we change css etc.

**Comment by tomholub** [comment]
I'm ok with it as is, given our options

**Comment by tomholub** [comment]
Meaning, I'd be ok to merge it

File: test/source/tests/tests/gmail.ts (hunk 1 above)
Comment (limonte) [reply test/source/tests/tests/gmail.ts]: > That I think would be too too noisy - when we change css etc.

Yeah, agree.

**Comment by limonte** [comment]
> Meaning, I'd be ok to merge it

Go ahead :+1:

**Pull request merged**